sequence_processing_pipeline/NuQCJob.py

```diff
@@ -394,15 +458,26 @@ def _generate_mmi_filter_cmds(self, working_dir):
 
         cores_to_allocate = int(self.cores_per_task / 2)
 
-        if len(self.additional_fastq_tags) > 0:
+        # contains_bx_metadata() will scan all of the R1 and R2 files
+        # in self.root_dir until it finds a non-empty file to read. It will
+        # read the first line of the compressed fastq file and see if it
+        # contains optional BX metadata. If not it will return False, other
+        # wise True.
+        apply_bx = self.contains_bx_metadata()
+
+        # the default setting.
+        tags = ""
+        t_switch = ""
+
+        if apply_bx & len(self.additional_fastq_tags) > 0:
```

Comment:
Unclear why this is dependent on apply_bx. Shouldn't this branch be taken regardless if additional_fastq_flags is not empty?